src/lib.rs

```diff
@@ -67,21 +82,77 @@ pub use poll::{Poll, Registry};
 pub use token::Token;
 pub use waker::Waker;
 
-#[cfg(unix)]
-cfg_any_os_util! {
-    pub mod unix {
-        //! Unix only extensions.
-        pub use crate::sys::SourceFd;
-    }
+#[cfg(all(unix, feature = "os-util"))]
+pub mod unix {
+    //! Unix only extensions.
+    pub use crate::sys::SourceFd;
+}
+
+// Enable with `cargo doc --features extra-docs`.
+#[cfg(feature = "extra-docs")]
+pub mod features {
+    //! # Mio's optional features.
+    //!
+    //! This document describes the available features in Mio.
+    //!
+    #![cfg_attr(feature = "os-poll", doc = "## `os-poll` (enabled)")]
+    #![cfg_attr(not(feature = "os-poll"), doc = "## `os-poll` (disabled)")]
+    //!
+    //! Mio by default provides only a shell implementation, that `panic!`s the
+    //! moment it is actually run. To run it requires OS support, this is
```

Comment:
how about: `.. it is actually run. OS support is required in order to run and is enabled by activating the `os-poll` feature`.